src/stcal/saturation/saturation.py

```diff
@@ -85,6 +84,20 @@ def flag_saturated_pixels(
             plane = data[ints, group, :, :]
 
             if read_pattern is not None:
+                if group == 2:
```

Comment:
I don't think this code will work here;  it's looking for where SATURATED is set in groupdq for group==2, but the flags aren't passed from flag_temp into groupdq until after this code block executes.

Reply:
flag_temp? I don't see that used anywhere in here. Where is it?

Reply:
Sorry, copy-paste issue from my review of the jwst PR.  Same general idea though- gdq for the third group isn't set until the code block after all the (if group==2) stuff, so it can never fulfill the criterion of identifying things where saturation was set in the third group.